### DISCUSSION TITLE
Difficulties with solving Absolute Loss LP

### DISCUSSION THREAD

**Discussion by jeziddah** [yalmip/YALMIP#1075] [Optimization modelling] [answered]
Hello, 
I am following a course on Machine Learning: Linear Regression taught some years ago and I came across the attached problem. I have tried to implement the solution in MATLAB using both the attached MATLAB and  YALMIP codes keep getting error messages regarding  the matrix dimension. Is anyone here familiar with this problem and would be kind to offer some guidance?
<img width="482" alt="MATLAB and YALMIP Code" src="https://user-images.githubusercontent.com/58339113/153203611-48f3df69-e2a7-4299-896c-10bd64f96d77.png">
<img width="482" alt="Problem" src="https://user-images.githubusercontent.com/58339113/153203619-faddfe28-6bc1-4e2a-a01d-4b52e4e0c4c6.png">

**Comment by johanlofberg** [comment, marked as answer]
Huh? There are no errors showing in your pictures.

solvesdp is obsolete. The command is called optimize

**Reply by jeziddah** [reply]
Thank you Johan for your comment. please see attached the error message 
<img width="611" alt="Screenshot 2022-02-09 at 14 01 47" src="https://user-images.githubusercontent.com/58339113/153206490-45657a92-1161-4bfd-80ee-e12e45500be4.png">

**Reply by johanlofberg** [reply]
As MATLAB is telling  you, the dimensions simply don't match. Have you looked at what values n and m you have, the size of Phi and theta, and related that to the dimension of y

**Reply by jeziddah** [reply]
Yes, I have and indeed they are not related. pls see attached the sizes.
<img width="278" alt="Screenshot 2022-02-09 at 14 14 36" src="https://user-images.githubusercontent.com/58339113/153208687-aab4c54a-0f16-42c6-9c3c-19cbebc4a4c6.png">

**Reply by johanlofberg** [reply]
Hence fix that. You cannot ask MATLAB to magically survive and not crash when you are trying to do things that don't make sense

**Reply by jeziddah** [reply]
May I ask if the problem has to do with the solution or the coding of the solution

**Reply by johanlofberg** [reply]
For instance, you are trying to define Phi*theta-y. The matrix Phi has width 2 and y has width 1, hence what dimension must theta have for that expression to make sense?

**Reply by jeziddah** [reply]
Theta must have the same dimension as Phi , in that case 2

**Reply by johanlofberg** [reply]
So why don't you have that then...

**Reply by jeziddah** [reply]
Honestly, I don't know what to do at the moment to solve the problem. 
The same problem appeared in Pro Stefano Ermon @ermongroup lesson on Machine learning.

**Reply by johanlofberg** [reply]
You've concluded that using n=1 is wrong, and the correct value should be n=2, but you still say you don't know what to do?

**Reply by johanlofberg** [reply]
and don't use sum(abs(x)), much faster in YALMIP to use norm(x,1)

**Reply by jeziddah** [reply]
> You've concluded that using n=1 is wrong, and the correct value should be 2, but you still say you don't know what to do?

I tried making theta as the same length as phi but i still get the same error

**Reply by johanlofberg** [reply]
No you don't, you are doing something else. If theta has length 2, the multiplication Phi * theta cannot fail, and neither can Phi * theta-y

**Reply by jeziddah** [reply]
I am following the solution provided in the lecture as attached to this discussion. however, I noticed that by implementing the solution to the problem, the matrix and the vector dimensions are not related as I attached earlier. So I can understand the error message. Since its a new area I am venturing into, I am struggling to figure it out.  the more I search about it, the more I come across materials with the same solution that returns the attached error message.

**Reply by johanlofberg** [reply]
But as you've already concluded, the computation of n in the code is incorrect as it is based on the wrong matrix.
